lib/rules/complexity.js

```diff
@@ -11,8 +11,15 @@
 //------------------------------------------------------------------------------
 
 module.exports = function(context) {
+    var option = context.options[0],
+        THRESHOLD = 20;
 
-    var THRESHOLD = (typeof context.options[0] !== "undefined") ? context.options[0] : 20;
+    if (option.hasOwnProperty("maximum") && typeof option.maximum === "number") {
+        THRESHOLD = option.maximum;
+    }
+    if (typeof option === "number") {
+        THRESHOLD = option;
+    }
```

Comment:
We don't have a good convention for sharing code between rules right now. We discussed it back in #848, but that didn't really go anywhere.